### PR TITLE
drop support for Python 3.8

### PR DESCRIPTION
**Pull request by zacharyburnett**
Numpy is imminently dropping support for Python 3.8

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -11,9 +11,13 @@ sphinx:
   configuration: docs/conf.py
   fail_on_warning: true
 
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
+
```
Comment (WilliamJamieson) [inline .readthedocs.yaml:18]: The RTD configuration should be redone to use mamba, see https://github.com/astropy/astropy/blob/main/.readthedocs.yaml for an example

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]

File: .readthedocs.yaml [hunk 2]
```diff
@@ -11,9 +11,13 @@ sphinx:
   configuration: docs/conf.py
   fail_on_warning: true
 
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
```
Comment (zacharyburnett) [inline .readthedocs.yaml:17]:
```suggestion
    python: "mambaforge-4.10"
```

**Comment by zacharyburnett** [comment]
It looks like the docs build is failing with the following:
https://readthedocs.org/projects/roman-datamodels/builds/20119428/
```
/home/docs/checkouts/readthedocs.org/user_builds/roman-datamodels/checkouts/155/docs/roman_datamodels/datamodels/developer_api.rst:113: WARNING: autosummary: stub file not found 'roman_datamodels.stuserdict.Any'. Check your autosummary_generate setting.
```

**Comment by nden** [comment]
@zacharyburnett I wonder if this helps
https://stackoverflow.com/questions/47344477/how-to-exclude-imports-from-automodapi-output/68984608#68984608

**Comment by nden** [comment]
@zacharyburnett This needs a change log entry

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]
- https://github.com/spacetelescope/roman_datamodels/blob/ae1caffaa0cccebdbfa2ac41907c8e6717d52ad2/src/roman_datamodels/stnode.py#L27-L30 should be fixed as this is for python 3.8 support
- Fix merge conflicts.

**Review by PaulHuwe — APPROVED** [review]
LGTM

**Comment by nden** [comment]
@zacharyburnett There's now an ImportError. Once this is fixed we can merge.

> "datamodels-1.0.yaml"
> E   NameError: name 'importlib_resources' is not defined

**Review by nden — APPROVED** [review]

**Comment by nden** [comment]
The change request was addressed, merging.

**Pull request merged**